packages/adp-tooling/src/types.ts

```diff
@@ -35,21 +35,41 @@ export interface AdpWriterConfig {
     target: AbapTarget;
     ui5?: {
         minVersion?: string;
+        version?: string;
+        url?: string;
     };
     package?: {
         name?: string;
         description?: string;
     };
+    flp?: {
+        title?: string;
+        subtitle?: string;
+        inboundId?: string;
+        bspName?: string;
+        languages?: Language[];
+        credentials?: {
+            username: string;
+            password: string;
+        };
+    };
+    customConfig?: AdpCustomConfig;
     /**
      * Optional: configuration for deployment to ABAP
      */
-    deploy?: Adp;
+    deploy?: Adp | BspApp;
     options?: {
         /**
          * Optional: if set to true then the generated project will be recognized by the SAP Fiori tools
          */
         fioriTools?: boolean;
     };
+    appdescr?: ManifestAppdescr;
```

Comment:
As discussed on the phone, please reduce this to a change array (i.e. the content section of the appdescr) so that the caller can push additional changes but cannot affect the rest of the file.

Reply:
Done